doc/main/reference/try_put_and_wait.rst

```diff
@@ -0,0 +1,324 @@
+.. _try_put_and_wait:
+
+Waiting for single messages in Flow Graph
+=========================================
+
+.. contents::
+    :local:
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` to the receiving nodes in the Flow Graph.
+This function allows to wait until completion of single message, submitted to ``try_put_and_wait`` as an input by the Flow Graph in
+opposite to ``graph.wait_for_all()`` function that waits for completion of all tasks in the graph.
+
+The ``node.try_put_and_wait(msg)`` is performing ``node.try_put(msg)`` on the node and blocks until the work on ``msg`` is completed.
+It means the following two statements:
+
+* Any task spawned by any node in the Flow Graph that corresponds on working with ``msg`` or any other intermediate result
+computted from ``msg`` is completed.
+* Any intermediate result computted from ``msg`` is retrieved from any buffer in the graph.
+
+.. caution::
+
+    Usage of buffering nodes at the end of the Flow Graph to collect the computation results may prevent ``try_put_and_wait`` calls
+    even after ``wait_for_all`` call would be completed.
+
+.. caution::
+
+    ``multifunction_node`` and ``async_node`` classes are not currently supported by this feature. Having one of these nodes in the
```

Comment:
The ``multifunction_node`` and ``async_node`` classes are not supported by this feature. Including one of these nodes in the Flow Graph may cause ``try_put_and_wait`` to exit early, even if the computations on the initial input message are still in progress.

Reply:
Fixed.